service/history/ndc/workflow_state_replicator.go

```diff
@@ -526,7 +526,8 @@ func (r *WorkflowStateReplicatorImpl) applySnapshot(
 			return err
 		}
 	} else {
-		err = r.taskRefresher.PartialRefresh(ctx, localMutableState, localTransitionHistory[len(localTransitionHistory)-1])
+		// Don't also refresh state machine tasks, that's done in mutable state ApplySnapshot already.
```

Comment:
Those tasks are discarded via the PopTasks() call on L507, so we can solely rely on PartialRefresh.

Actually it should be ApplySnapshot that does the PopTasks, not the caller. cc @xwduan